ocf_datapipes/transform/numpy/encode_space_time.py

```diff
@@ -0,0 +1,204 @@
+import warnings
+from numbers import Number
+
+import numpy as np
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+from ocf_datapipes.consts import BatchKey
+from ocf_datapipes.utils import NumpyBatch
+
+
+@functional_datapipe("encode_space_time")
+class EncodeSpaceTimeIterDataPipe(IterDataPipe):
+    def __init__(
+        self,
+        source_dp: IterDataPipe,
+        lengths: dict[str, Number] = lambda: dict(
+            x_osgb=480_000, y_osgb=400_000, time_utc=60 * 5 * 37
+        ),
+        n_fourier_features_per_dim: int = 8,
+    ):
+        self.source_dp = source_dp
+        self.lengths = lengths
+        self.n_fourier_features_per_dim = n_fourier_features_per_dim
+
+    def __iter__(self):
+        for np_batch in self.source_dp:
+            yield get_spatial_and_temporal_fourier_features(
+                np_batch=np_batch,
+                lengths=self.lengths,
+                n_fourier_features_per_dim=self.n_fourier_features_per_dim,
+            )
+
+
+def get_spatial_and_temporal_fourier_features(
+    np_batch: NumpyBatch,
+    lengths: dict[str, Number],
+    n_fourier_features_per_dim: int = 8,
+) -> NumpyBatch:
+    """Add fourier features for x_osgb, y_osgb and time_utc."""
+
+    rescaled_coords: dict[str, np.ndarray] = _rescale_coords_for_all_dims_to_approx_0_to_1(
```

Comment:
this rescales depending on the mininimum in the batch, should we be using a global minumum and maybe maximum? Perhaps i havent fully undrestoof what these fourier features do. But just thinking about normalizing data int he dataloader, we use a global value, and not a batch value